bagelbids/tests/test_cli_bids.py

```diff
@@ -42,10 +84,42 @@ def test_bids_valid_inputs_run_successfully(
     ],
 )
 def test_check_unique_bids_subjects_err(bids_list, expectation):
-    """Given a list of BIDS subject IDs, raise an error or not depending on whether all IDs are found in the phenotypic subject list."""
+    """
+    Given a list of BIDS subject IDs, raise an error or not depending on
+    whether all IDs are found in the phenotypic subject list.
+    """
     pheno_list = ["sub-01", "sub-02", "sub-03", "sub-PD123", "sub-PD234"]
 
     with expectation:
         check_unique_bids_subjects(
             pheno_subjects=pheno_list, bids_subjects=bids_list
         )
+
+
+@pytest.mark.parametrize(
+    "bids_dir, acquisitions, bids_session",
+    [
+        (
+            "synthetic",
+            ["nidm:T1Weighted"] + ["nidm:FlowWeighted"] * 3,
+            "01",
+        ),
+        (
+            "ds001",
+            ["nidm:T2Weighted", "nidm:T1Weighted"] + ["nidm:FlowWeighted"] * 3,
+            None,
+        ),
+        ("eeg_ds000117", ["nidm:T1Weighted"], None),
+    ],
+)
+def test_create_acquisitions(bids_path, bids_dir, acquisitions, bids_session):
+    """Given a BIDS dataset, creates a list of acquisitions matching the image files found on disk."""
+    image_list = create_acquisitions(
+        layout=BIDSLayout(bids_path / bids_dir, validate=True),
+        bids_sub_id="01",
+        session=bids_session,
+    )
+
+    assert len(acquisitions) == len(image_list)
+    for i, image in enumerate(image_list):
+        assert acquisitions[i] == image.hasContrastType.identifier
```

Comment:
why is the ordering important here? In other words: if you expect there to be 3 `""nidm:T1Weighted"` contrastType acquisitions, why should it matter if they are also in the expected order in the `image_list`? Unless I misunderstand the intent here, a better assertion might be to say: 

```python
from collections import Counter

acquisitions = {
"nidm:T1Weighted": 1,
"nidm:FlowWeighted": 3
}

image_counts = Counter([image.hasContrastType.identifier for img in image_list])

for contrast, count in acquisitions.items():
  assert image_counts[contrast] == count

Reply:
The ordering isn't really important within a session, you're right. I think this assertion should achieve the same thing